src/CharsetDetector.cs

```diff
@@ -326,46 +357,46 @@ private void FindInputState(byte[] buf, int len)
                 else
                 {
                     if (InputState == InputState.PureASCII &&
-                        (buf[i] == 0x1B || (buf[i] == 0x7B && _lastChar == 0x7E)))
+                        (buf[offset + i] == 0x1B || (buf[offset + i] == 0x7B && _lastChar == 0x7E)))
                     {
                         // found escape character or HZ "~{"
                         InputState = InputState.EscASCII;
                         _escCharsetProber = _escCharsetProber ?? GetNewProbers();
                     }
-                    _lastChar = buf[i];
+                    _lastChar = buf[offset + i];
                 }
             }
         }
 
-        private static string FindCharSetByBom(byte[] buf, int len)
+        private static string FindCharSetByBom(byte[] buf, int offset, int len)

```

Comment:
Could you give an example when this is necessary? As far as I know, _magic number_ is inserted first

Reply:
The new method overload has an assumption: 'the actual data starts at buf[offset]'. It is useful when the actual data was loaded into the middle of the byte array.